frontend/src/tests/posting-tests.js

```diff
@@ -1,39 +1,354 @@
-import { it, beforeAll, apiGET, apiPOST } from "./utils";
+import {
+    it,
+    beforeAll,
+    expect,
+    checkPropTypes,
+    postingPropTypes,
+    postingPositionPropTypes,
+    addSession,
+} from "./utils";
+import PropTypes from "prop-types";
 import { databaseSeeder } from "./setup";
 
-// TODO: Remove eslint disable. This can be done as soon as these tests are actually implemented.
-// eslint-disable-next-line
-export function postingTests(api = { apiGET, apiPOST }) {
+export function postingTests(api) {
+    const { apiGET, apiPOST } = api;
+    let session;
+    let position;
+    let resp;
+    const posting = {
+        name: "CSC494F TA",
+        intro_text: "TA posting for CSC494F",
+        open_date: "2021/01/01",
+        close_date: "2021/05/01",
+        availability: "auto",
+        custom_questions: {
+            pages: [
+                {
+                    name: "page1",
+                    elements: [
+                        {
+                            type: "text",
+                            name: "question1",
+                        },
+                        {
+                            type: "text",
+                            name: "question2",
+                        },
+                    ],
+                },
+                {
+                    name: "page2",
+                    elements: [
+                        {
+                            type: "text",
+                            name: "question3",
+                        },
+                        {
+                            type: "text",
+                            name: "question4",
+                        },
+                    ],
+                },
+            ],
+        },
+    };
+    const postingPos = {
+        hours: 20,
+        num_positions: 1,
+    };
+
     beforeAll(async () => {
         await databaseSeeder.seed(api);
+        session = databaseSeeder.seededData.session;
+        position = databaseSeeder.seededData.position;
+    });
+
+    it("Create a posting for a session", async () => {
+        resp = await apiPOST("/admin/postings", {
+            ...posting,
+            session_id: session.id,
+        });
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(postingPropTypes, resp.payload);
+        expect(resp.payload.id).not.toBeNull();
+        Object.assign(posting, resp.payload);
+
+        const newPosting2 = {
```

Comment:
you may want to change the naming of `newPosting2`, as there's only one new `Posting` for this test